scrape/manager.go

```diff
@@ -223,10 +223,18 @@ func (m *Manager) TargetsActive() map[string][]*Target {
 	m.mtxScrape.Lock()
 	defer m.mtxScrape.Unlock()
 
+	var wg sync.WaitGroup
 	targets := make(map[string][]*Target, len(m.scrapePools))
 	for tset, sp := range m.scrapePools {
-		targets[tset] = sp.ActiveTargets()
+		wg.Add(1)
```

Comment:
You can have a single `wg.Add(len(m.scrapePools))` call before the `for` loop.

Reply:
Done.